bin/modules/untappd/untapped_module.dart

```diff
@@ -0,0 +1,258 @@
+  Future<bool> _isValidUsername(String untappdUsername) async {
+    final webScraper = WebScraper('https://untappd.com');
+    if (await webScraper.loadWebPage('/user/$untappdUsername')) {
+      final checkins = webScraper.getElementAttribute(
+          'div#main-stream > *', 'data-checkin-id');
+
+      if (checkins.isEmpty) {
+        return false;
+      }
+      return true;
+    } else {
+      throw 'Error during fetching of Untappd data';
+    }
+  }
+
+  /// Get latest checkin for given untapped username
+  Future<UntappdCheckin?> _getLatestCheckin(String untappdUsername) async {
+    final webScraper = WebScraper('https://untappd.com');
+    if (await webScraper.loadWebPage('/user/$untappdUsername')) {
+      final checkins = webScraper.getElementAttribute(
+          'div#main-stream > *', 'data-checkin-id');
+
+      if (checkins.isEmpty) {
+        throw 'No checkins are available for $untappdUsername';
+      }
+
+      var latestCheckin = checkins.first!;
+
+      var baseCheckinAddress =
+          'div#main-stream > #checkin_$latestCheckin > div.checkin > div.top';
+
+      final checkinTitleElement =
+          webScraper.getElementTitle('$baseCheckinAddress > p.text');
+      final checkinTitle =
+          checkinTitleElement.isEmpty ? '' : checkinTitleElement.first.trim();
+
+      final checkinRatingElement = webScraper.getElement(
+          '$baseCheckinAddress > div.checkin-comment > div.rating-serving > div.caps ',
+          ['data-rating']);
+      final String checkinRating = checkinRatingElement.isEmpty
+          ? '0'
+          : checkinRatingElement.first['attributes']['data-rating'];
+
+      final checkinCommentElement = webScraper.getElementTitle(
+          '$baseCheckinAddress > div.checkin-comment > p.comment-text');
+      final checkinComment = checkinCommentElement.isEmpty
+          ? ''
+          : checkinCommentElement.first.trim();
+
+      final photo = webScraper.getElementAttribute(
+          '$baseCheckinAddress > p.photo > a > img', 'data-original');
+      final checkinPhotoAddress = photo.isNotEmpty ? photo.first : null;
+
+      return UntappdCheckin(
+          id: latestCheckin,
+          title: checkinTitle,
+          rating: checkinRating,
+          comment: checkinComment,
+          photoAddress: checkinPhotoAddress);
+    }
+    return null;
+  }
+
+  /// Get untappd detailed checkin URL
+  String _getCheckinUrl(String checkinId, String username) =>
+      'https://untappd.com/user/$username/checkin/$checkinId';
+
+  @override
+  void init(INyxxWebsocket bot,
+      {Duration updateInterval = const Duration(minutes: 12)}) {
+    // Set up Hive for local data storage
+    Hive.init('/data');
+    Hive.openBox(HiveConstants.untappdBox);
+
+    _bot = bot;
+
+    // Start timer to check for untappd updates
+    Timer.periodic(updateInterval, (timer) => _checkUntappd());
+
+    // Set module as initialized
+    _isInitialized = true;
```

Comment:
**Suggestion:** Use dependency injection to provide the `INyxxWebsocket bot` to the `UntappdModule` instead of setting it in the `init` method. This will make the code more testable and modular.
```suggestion
  UntappdModule(INyxxWebsocket bot) {
  ...
  _bot = bot;
  ...
  }
```